src/ui/NoNameFacet/NoNameFacetHeader.ts

```diff
@@ -88,23 +92,66 @@ export class NoNameFacetHeader {
   }
 
   private recreateOperator() {
-    $$(this.operatorElement).replaceWith(this.createOperator());
+    const newElement = this.createOperator();
+    $$(this.operatorElement).replaceWith(newElement);
+    this.operatorElement = newElement;
+  }
+
+  private createCollapse() {
```

Comment:
I think the `collapse` button/component can live in it's own file. Aside from separating concerns, it makes the logic easier to test, and gives shorter and less nested test files. It also naturally defines what methods should be `public` vs. `private` based on how the importing file wants to interact with the component.

In addition, it makes reuse of the component easier.

I think the existing Facet separated a lot of different parts well, but it would have benefited even more from further segmentation. For the new series of Facets, it would be a good goal to aim to minimise the amount of copy-pasting, and having many modular components.

Reply:
you're right I'll think of something